lesson_7/oana_agafitei/lesson7_decorators_homework.py

```diff
@@ -0,0 +1,57 @@
+
+'''def uppercase(name):
+
+    def wrapper(input):
+
+        res = name(input)
+        modified = res.upper()
+
+        return modified
+    return wrapper
+
+@uppercase
+def greet(name):
+    return "Greetings {}!".format(name)
+
+print(greet("World"))'''
+
+
+
+def safe_divide(func):
+    def inside(numerator, denominator):
+        if denominator==0:
+            print("Can't divide by 0")
+            return
+        func(numerator, denominator)
+    return inside
+
+@safe_divide
+def divide(first_number, second_number):
+    return first_number / second_number
+
+print(divide(10, 0))
+
+
+
+print_registry = []
+
+def register(func):
+    global print_registry
```

Comment:
in cazul acesta nu avem nevoie de `global` print_registry este deja intr-un scop superior si poate fi folosit